eng/common/scripts/SetTestPipelineVersion.ps1

```diff
@@ -19,10 +20,13 @@ Foreach ($tags in $latestTags)
   $semVars += $tags.Replace("${PackageName}_", "")
 }
 
-$semVarsSorted = [AzureEngSemanticVersion]::SortVersionStrings($semVars)
-LogDebug "Last Published Version $($semVarsSorted[0])"
+if ($semVars -and $semVars.Length -gt 0)
+{
+  $semVarsSorted = [AzureEngSemanticVersion]::SortVersionStrings($semVars)
+  LogDebug "Last Published Version $($semVarsSorted[0])"
+  $newVersion = [AzureEngSemanticVersion]::new($semVarsSorted[0])
```

Comment:
Won't $newVersion be null still if semVars is null/empty?

Reply:
The default of 1.0.0 isn't good. Running the changed script locally, in branch of the Java repository that's hitting the null error, ends up with "Version to publish [ 1.0.0 ]" which is what it sets my test library to. This is setting it to a GA version which is not okay. This should be using pre-release versions and the buildID that's being passed in. I'd expect 1.0.0-beta.BuildId

Reply:
@chidozieononiwu, there's a secondary reason why 1.0.0 instead of 1.0.0-beta.BuildID wouldn't work. In Java you cannot release a GA version with beta dependencies. There are 3 template libraries (azure-sdk-template, azure-sdk-template-two and azure-sdk-template-three) that depend on each other (two depends on the azure-sdk-template, three depends on two and azure-sdk-template). This would cause a release version check failure, it needs to be beta.BuildID as a default.